src/store/staking/staking.sagas.ts

```diff
@@ -128,13 +131,61 @@ export function* fetchStakesList() {
   }
 }
 
+export function* fetchFilteredHistoryStakeEvents(contractAddresses: string[]) {
+  const subgraphClient = yield* select(subgraphClientSelector);
+
+  if (!subgraphClient) throw new Error('Wallet not connected');
+
+  const { stakeEvents } = yield* call(historyStakesQuery, subgraphClient, {
+    contractAddresses,
+  });
+  return stakeEvents;
+}
+
+export function* getStakesAndVestsAddresses() {
+  const account = yield* select(accountSelector);
+  const vestsList = yield* select(vestsListSelector);
+
+  if (!account || !vestsList) {
+    throw new Error('Wallet not connected');
+  }
+
+  const addresses = vestsList.map((vest) => vest.address);
+  addresses.push(account.toLowerCase());
+  return addresses;
+}
+
+export function* fetchHistoryStakingForContract() {
```

Comment:
Actually we're not fetching it "for contract" but for user and multiple vesting contracts :)
```suggestion
export function* fetchHistoryStaking() {
```